From f5973bc298b6420456b762b1506f87c2aa524d8b Mon Sep 17 00:00:00 2001
From: Charalampos Stratakis <cstratak@redhat.com>
Date: Tue, 5 Aug 2025 16:00:47 +0200
Subject: [PATCH] Replace __pyx_empty_tuple with PyTuple_New(0)

__pyx_empty_tuple wasn't part of Cython's public API and
the underlying mechanism for generating an empty tuple changed
with Cython 3.1.

This change ensures that the proper macro is used and makes
dionaea compile with Cython >= 3.1.

See also: https://github.com/cython/cython/pull/6351
---
 modules/python/module.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/python/module.h b/modules/python/module.h
index 726a7892..9b842df9 100644
--- a/modules/python/module.h
+++ b/modules/python/module.h
@@ -9,9 +9,9 @@
 #include <stdbool.h>
 #include "connection.h"
 
-#define PY_CLONE(T)  (T)->ob_type->tp_new((T)->ob_type, __pyx_empty_tuple, NULL)
-#define PY_NEW(T) (((PyTypeObject*)(T))->tp_new( (PyTypeObject*)(T), __pyx_empty_tuple, NULL))
-#define PY_INIT(P, O) (P)->ob_type->tp_init((O), __pyx_empty_tuple, NULL)
+#define PY_CLONE(T)  (T)->ob_type->tp_new((T)->ob_type, PyTuple_New(0), NULL)
+#define PY_NEW(T) (((PyTypeObject*)(T))->tp_new( (PyTypeObject*)(T), PyTuple_New(0), NULL))
+#define PY_INIT(P, O) (P)->ob_type->tp_init((O), PyTuple_New(0), NULL)
 #define REFCOUNT(T) printf("obj refcount %i\n", (int)(T)->ob_refcnt)
 
 #define REMOTE(C) (C)->remote
